### PR TITLE
ARROW-11965: [R][Docs] Simplify install.packages command in R dev docs

### PR DESCRIPTION
**Pull request by pachadotdev**
Hi

I have used the same setup that I used when I worked in finance, this is to call `R  --vanilla` from the terminal, and use the global CRAN  repository, so that this fix applies to Windows/Mac/Linux.

The changes made solve the following problems:
- Packages not installing because the default mirror is not found
- The lack of packages bit64 and tidyselect

Best,

### CODE REVIEW

File: r/README.md [hunk 1]
```diff
@@ -143,7 +143,15 @@ checkout:
 
 ``` shell
 cd ../../r
-R -e 'install.packages(c("devtools", "roxygen2", "pkgdown", "covr")); devtools::install_dev_deps()'
+
+R --vanilla << EOF
+options(repos = c(REPO_NAME = "https://cloud.r-project.org/"))
+install.packages(c("bit64", "devtools", "roxygen2", "tidyselect",
+ "pkgdown", "covr")
```
Comment (ianmcook) [inline r/README.md:150]: This is missing a `)`

Comment (ianmcook) [reply r/README.md:150]: Could you also please keep the `install.packages()` command all on one line?

File: r/README.md [hunk 2]
```diff
@@ -143,7 +143,15 @@ checkout:
 
 ``` shell
 cd ../../r
-R -e 'install.packages(c("devtools", "roxygen2", "pkgdown", "covr")); devtools::install_dev_deps()'
+
+R --vanilla << EOF
```
Comment (ianmcook) [inline r/README.md:147]: These instructions are intended for Windows users too (at least nominally) and this `<< EOF` heredoc trick does not work on Windows.

Comment (nealrichardson) [reply r/README.md:147]: I think it's also confusing; I'd go back to the one-liner, that wasn't the problem.

Comment (pachamaltese) [reply r/README.md:147]: I changed this to Rscript instead, thanks !!

File: r/README.md [hunk 3]
```diff
@@ -143,7 +143,15 @@ checkout:
...
+options(repos = c(REPO_NAME = "https://cloud.r-project.org/"))
+install.packages(c("bit64", "devtools", "roxygen2", "tidyselect",
```
Comment (nealrichardson) [inline r/README.md]: bit64 and tidyselect are hard dependencies of arrow so you don't need to install them explicitly

None of the other dependencies are required to build the package, and they bring a lot of stuff with them, so it's probably better to leave them off as a separate step.

The quickest version of this is `install.packages("remotes"); remotes::install_deps(dependencies = TRUE)`

File: r/README.md [hunk 4]
```diff
@@ -143,7 +143,15 @@ checkout:
 
 ``` shell
 cd ../../r
-R -e 'install.packages(c("devtools", "roxygen2", "pkgdown", "covr")); devtools::install_dev_deps()'
+
+R --vanilla << EOF
+options(repos = c(REPO_NAME = "https://cloud.r-project.org/"))
```
Comment (nealrichardson) [inline r/README.md:148]:
```suggestion
options(repos = "https://cloud.r-project.org/")
```

**Review by nealrichardson — CHANGES_REQUESTED** [review]

**Review by nealrichardson — APPROVED** [review]

**Pull request closed without merging**